src/Nethermind/Nethermind.Network/IPResolver.cs

```diff
@@ -58,57 +64,60 @@ public IpResolver(INetworkConfig networkConfig, ILogManager logManager)
 
         private IPAddress InitializeExternalIp()
         {
-            string externalIpSetInEnv = Environment.GetEnvironmentVariable("NETHERMIND_ENODE_IPADDRESS");
-            if (externalIpSetInEnv != null)
+            IEnumerable<IIPSource> GetIPSources()
             {
-                return IPAddress.Parse(externalIpSetInEnv);
-            }
-            
-            if (_networkConfig.ExternalIp != null)
-            {
-                if(_logger.IsWarn) _logger.Warn($"Using the external IP override: {nameof(NetworkConfig)}.{nameof(NetworkConfig.ExternalIp)} = {_networkConfig.ExternalIp}");
-                return IPAddress.Parse(_networkConfig.ExternalIp);
+                yield return new EnvironmentVariableIPSource();
+                yield return new NetworkConfigExternalIPSource(_networkConfig, _logManager);
+                yield return new WebIPSource("http://checkip.amazonaws.com", _logManager);
+                yield return new WebIPSource("http://icanhazip.com", _logManager);
+                yield return new WebIPSource("http://bot.whatismyipaddress.com", _logManager);
+                yield return new WebIPSource("http://ipinfo.io/ip", _logManager);
+                yield return new WebIPSource("http://api.ipify.org", _logManager);
             }
             
             try
             {
-                const string url = "http://checkip.amazonaws.com";
-                if(_logger.IsInfo) _logger.Info($"Using {url} to get external ip");
-                string ip = new WebClient().DownloadString(url).Trim();
-                if(_logger.IsDebug) _logger.Debug($"External ip: {ip}");
-                ThisNodeInfo.AddInfo("External IP  :", $"{ip}");
-                
-                return IPAddress.Parse(ip);
+                foreach (var s in GetIPSources())
+                {
+                    if (s.TryGetIP(out var ip))
+                    {
+                        ThisNodeInfo.AddInfo("External IP  :", $"{ip}");
+                        return ip;
+                    }
+                }
             }
             catch (Exception e)
```

Comment:
since we catch Exception here, we can catch it in sources too

Reply:
Other exceptions are handled on higher level as they are actually not even expected here